supervisor/docker/addon.py

```diff
@@ -469,12 +476,13 @@ def _run(self) -> None:
                 self.image,
                 tag=str(self.addon.version),
                 name=self.name,
-                hostname=self.addon.hostname,
+                hostname=None if self.uts_mode else self.addon.hostname,
```

Comment:
So we're telling the container it has no hostname but we're still telling the DNS plugin it does here:
https://github.com/home-assistant/supervisor/blob/716885d0475fdd410176f880ae54de0bb4a7d2c3/supervisor/docker/addon.py#L513
I get that addons with `host_uts` can see the hostname of the host but why does that also mean they don't have their own? Think I'm not understanding exactly what this does for an addon.

Reply:
> So we're telling the container it has no hostname 

Not exaclty, it seems that the API needs `None` if we don't have our own UTS namesapce. If we try to set a hostname, and don't have our own, the API returns with an exception from what I remember.

> I get that addons with host_uts can see the hostname of the host but why does that also mean they don't have their own? 

That is not really a "can see" thing, its really an identity thing. Using the very low level [`gethostname`](https://man7.org/linux/man-pages/man2/gethostname.2.html) returns the hostname of the underlying system instead of a "virtual" host name of your container's UTS namespace.

But agreed, adding the hostname to the DNS system is not required in this case. The underlying host should already resolve.

Reply:
Sorry, missed that comment.

It is essentially the same as if multiple processes running on the same host: If they fight over control of the hostname, then it gets problematic :D

On Home Assisant OS, the hostname is set by systemd. On init.d based system its typically some init daemon setting the hostname. Neither of those should run in a container! In a container, the hostname is setup by Docker itself.

So: No, this shouldn't be problematic, unless add-ons do nasty things (which they shouldn't anways). Also, they only can change if they have `CAP_SYS_ADMIN` as well. So of all the `host_*` options we have, this is probably one of the safer option :)